### PR TITLE
Lps 191375

### PR DESCRIPTION
**Pull request by manuelRives**
This is a reupload of #4608.

### CODE REVIEW

**Comment by manuelRives** [comment]
ci:test:sf

**Comment by manuelRives** [comment]
ci:test:relevant

**Comment by manuelRives** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java [hunk 1]
```diff
@@ -69,18 +70,69 @@ public void testHasFilesWithFileEntryFriendlyURLSeparator()
...
+		String nameBase = RandomTestUtil.randomString();
+
+		String fileURL = nameBase + "%2B.txt";
+		String fileName = nameBase + "+.txt";
+
+		Assert.assertFalse(
+			WebServerServlet.hasFiles(
+				_createMockHttpServletRequest(
+					String.format("/%s/0/%s", group.getGroupId(), fileURL))));
+		_addFileEntry(fileName, RandomTestUtil.randomString());
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java:84]: You need to add a blank line between the assertion and the call to `_addFileEntry`.

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java [hunk 2]
```diff
@@ -69,18 +70,69 @@ public void testHasFilesWithFileEntryFriendlyURLSeparator()
...
+
+		Assert.assertTrue(
+			WebServerServlet.hasFiles(
+				_createMockHttpServletRequest(
+					String.format("/%s/0/%s", group.getGroupId(), fileURL))));
+
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java:90]: You can remove this blank line.

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java [hunk 3]
```diff
@@ -69,18 +70,69 @@ public void testHasFilesWithFileEntryFriendlyURLSeparator()
...
+		Assert.assertTrue(
+			WebServerServlet.hasFiles(
+				_createMockHttpServletRequest(
+					String.format("/%s/0/%s", group.getGroupId(), fileURL))));
+		Assert.assertTrue(
+			WebServerServlet.hasFiles(
+				_createMockHttpServletRequest(
+					String.format(
+						"/%s/0/%s", group.getGroupId(),
+						nameBase + "%20.txt"))));
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/webserver/test/WebServerFriendlyURLTest.java:119]: Same formatting issue as in my previous comment.

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java [hunk 4]
```diff
@@ -1476,6 +1473,15 @@ private static Group _getGroup(long companyId, String name)
 		return user.getGroup();
 	}
 
+	private static String _getPath(HttpServletRequest httpServletRequest) {
+		String path = httpServletRequest.getRequestURI();
+		String location =
+			httpServletRequest.getContextPath() +
+				httpServletRequest.getServletPath();
+
+		return path.substring(location.length() + 1);
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java:1482]: Shouldn't `path` be named `requestURI` and `location` be renamed to `path`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java (hunk 4 above)
Comment (manuelRives) [reply portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java]: Completely agree. I just updated it. Thanks!.

**Comment by manuelRives** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by WYZ-Issac** [comment]
Failures are unrelated. Forward directly.

**Comment by WYZ-Issac** [comment]
ci:forward:force

**Pull request closed without merging**